### DISCUSSION TITLE
Graphic RAM buffer option

### DISCUSSION THREAD

**Discussion by TheWildJarvi** [hneemann/Digital#776] [Q&A] [answered]
Hi, I am interested in using the graphic ram to render some 3D graphics. I would like to be able to know if the B input supports what I would like to do.

I would like to draw a cube1 in buffer 1, while displaying buffer 0. 
when the frame is drawn in buffer 1, the screen switches to displaying buffer 1, and the render unit starts rendering cube 2 in buffer 0
etc.

Is this how the buffer input works?

jarvi

**Comment by hneemann** [comment, marked as answer]
Yes, if you set the size of the graphic, the memory will have exactly twice the size to store two images. The images follow each other directly in the memory. The input B then determines which image is displayed.
It is a classic double buffering approach.

**Comment by TheWildJarvi** [comment]
I'm unsure how to use it though. Do I manually have to double the amount of addresses in memory and apply my own offset? Then how does the B input know where the start of the first or second buffer is? On Jun 17, 2021 2:46 PM, Helmut Neemann ***@***.***> wrote:
Yes, if you set the size of the graphic, the memory will have exactly twice the size to store two images. The images follow each other directly in the memory. The input B then determines which image is displayed.
It is a classic double buffering approach.

**Reply by hneemann** [reply]
Each pixel is stored in a single word.
When width (dx) and height (dy) are set, dx\*dy is the size of a frame buffer. And the size of the memory is set to dx\*dy\*2. The offset of the first frame buffer is 0 and the offset of the second frame buffer is dx\*dy.

**Reply by TheWildJarvi** [reply]
This makes a lot of sense, this is why it always forces me to add an extra bit to my address, Thank you a lot!
